src/Microsoft.AspNet.Mvc.Core/Controller.cs

```diff
@@ -376,6 +376,11 @@ public virtual ContentResult Content(string content, string contentType, Encodin
         [NonAction]
         public virtual JsonResult Json(object data)
         {
+            var disposableValue = data as IDisposable;
+            if (disposableValue != null)
+            {
+                Response.OnResponseCompleted((f) => disposableValue.Dispose(), null);
```

Comment:
Use `_` as the param for the lambda if it's not really needed (and remove the extra parens).